core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -439,9 +467,47 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
         m_league_size(arg_policy.league_size()),
         m_team_size(arg_policy.team_size()),
         m_vector_size(arg_policy.impl_vector_length()) {
-    // FIXME_SYCL check_parameters
     // FIXME_SYCL optimize
     if (m_team_size < 0) m_team_size = 32;
+
+    // FIXME_SYCL modify for reduce etc.
+    m_shmem_begin = 0;
+    m_shmem_size =
+        (m_policy.scratch_size(0, m_team_size) +
+         FunctorTeamShmemSize<FunctorType>::value(m_functor, m_team_size));
+    m_scratch_size[0] = m_policy.scratch_size(0, m_team_size);
+    m_scratch_size[1] = m_policy.scratch_size(1, m_team_size);
+
+    // FIXME_SYCL so far accessors used instead of these pointers
+    // Functor's reduce memory, team scan memory, and team shared memory depend
+    // upon team size.
+    const auto& space    = *m_policy.space().impl_internal_space_instance();
+    const sycl::queue& q = *space.m_queue;
+    m_scratch_ptr[0]     = nullptr;
+    m_scratch_ptr[1]     = sycl::malloc_device(
+        sizeof(char) * m_scratch_size[1] * m_league_size, q);
+
+    if (static_cast<int>(space.m_maxShmemPerBlock) < m_shmem_size) {
+      std::stringstream out;
+      out << "Kokkos::Impl::ParallelFor<SYCL> insufficient shared memory! "
+             "Requested "
+          << m_shmem_size << " bytes but maximum is "
+          << m_policy.space().impl_internal_space_instance()->m_maxShmemPerBlock
+          << '\n';
+      Kokkos::Impl::throw_runtime_exception(out.str());
+    }
+
+    if (m_team_size > m_policy.team_size_max(arg_functor, ParallelForTag{}))
+      Kokkos::Impl::throw_runtime_exception(
+          "Kokkos::Impl::ParallelFor<SYCL> requested too large team size.");
+  }
+
+  ~ParallelFor() {
+    const Kokkos::Experimental::SYCL& space = m_policy.space();
+    Kokkos::Experimental::Impl::SYCLInternal& instance =
+        *space.impl_internal_space_instance();
+    sycl::queue& q = *instance.m_queue;
+    sycl::free(m_scratch_ptr[1], q);
```

Comment:
Please delete the copy constructor and copy assignment.  This is a bug otherwise.